### PR TITLE
Classification Module

### PR DESCRIPTION
**Pull request by cr458**
## Summary

  - Add **detection review mode UI** with probability-scaled colormap, threshold slider, and interactive labeling
  - Enable **iterative retraining** by accepting multiple training files via `--positives` with label preservation
  - Implement **sample weighting** for correction samples (`--correction-weight`) to emphasize relabeled detections
  - Add `--batch-size` CLI parameter for inference tuning
  - Add 13 new tests for classification module (combine_datasets, EmbeddingClassifier, detection mode)

  ## Features

  ### Detection Review Mode
  - Auto-detect detection GeoJSON (files with `probability` field)
  - Probability threshold slider with dynamic min/max from dataset range
  - Colormap styling: plasma gradient scaled to dataset probability range
  - Labeled detection colors: blue (positive), orange (negative)
  - Tile panel integration sorted by lowest probability first
  - Polygon labeling support for bulk corrections

  ### Iterative Retraining Workflow
  ```bash
  # Combine original labels with corrections
  uv run python -m geovibes.classification.pipeline \
      --positives original.geojson corrections.geojson \
      --output results/ \
      --db embeddings.duckdb \
      --correction-weight 2.5
  - combine_datasets() preserves labels from each file (no overwriting)
  - Correction samples use relabel_pos/relabel_neg class names
  - --correction-weight multiplies weights for correction samples during XGBoost training

  CLI Improvements

  | Argument            | Description                                                          |
  |---------------------|----------------------------------------------------------------------|
  | --positives         | Accept multiple GeoJSON files (nargs=+)                              |
  | --correction-weight | Weight multiplier for relabel_pos/relabel_neg samples (default: 1.0) |
  | --batch-size        | Batch size for inference (default: 100000)                           |

  Changed Files

  - geovibes/classification/pipeline.py - Multi-file support, sample weighting, CLI args
  - geovibes/classification/classifier.py - sample_weight parameter, removed unused ClassifierTiming
  - geovibes/ui/app.py - Detection mode UI, threshold controls, styling callbacks
  - geovibes/ui/map_manager.py - Detection layer with colormap styling
  - geovibes/ui/datasets.py - relabel_pos/relabel_neg class names for corrections
  - geovibes/ui/tiles.py - Detection mode label styling
  - tests/test_classification.py - New test file (13 tests)

  Test plan

  - combine_datasets() preserves labels from multiple files
  - combine_datasets() raises on missing label field
  - EmbeddingClassifier.fit() accepts sample_weight parameter
  - Model save/load preserves predictions
  - Detection mode state tracks tile_id -> label mapping
  - All 99 tests passing

### CODE REVIEW

**Comment by cr458** [comment]
@codex

**Comment by cr458** [comment]
@codex
